### PR TITLE
fix: timeout even when there's no message from the connection

### PR DESCRIPTION
**Pull request by 0xArdi**
## Fixes

Make sure the timeout gets checked even when there's no message from connection.
Backported from https://github.com/valory-xyz/mech/pull/74

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [x] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [x]  I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [x] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have locally run services that could be impacted and they do not present failures derived from my changes
- [x] Public-facing documentation has been updated with the changes affected by this PR. Even if the provided contents are not in their final form, all significant information must be included.
- [x] Any backwards-incompatible/breaking change has been clearly documented in the upgrading document.

### CODE REVIEW

**Review by angrybayblade — APPROVED** [review]

File: packages/valory/skills/abstract_round_abci/behaviour_utils.py [hunk 1]
```diff
@@ -357,6 +357,8 @@ def __handle_waiting_for_message(self) -> None:
                 # wait for the next message
                 self.__notified = False
                 self.__message = None
+        else:
+            self.__get_generator_act().send(None)
```
Comment (Adamantios) [inline packages/valory/skills/abstract_round_abci/behaviour_utils.py:361]: Cannot `StopIteration` also get raised here?

Comment (0xArdi) [reply packages/valory/skills/abstract_round_abci/behaviour_utils.py:361]: yes

Comment (Adamantios) [reply packages/valory/skills/abstract_round_abci/behaviour_utils.py:361]: Then how about:

```python
def __handle_waiting_for_message(self) -> None:
    """Handle an 'act' tick, when waiting for a message."""
    try:
        self.__get_generator_act().send(self.__message)
    except StopIteration:
        self.__handle_stop_iteration()
    finally:
        # wait for the next message
        self.__notified = False
        self.__message = None
```

File: packages/valory/skills/abstract_round_abci/behaviour_utils.py [hunk 2]
```diff
@@ -358,7 +358,10 @@ def __handle_waiting_for_message(self) -> None:
                 self.__notified = False
                 self.__message = None
         else:
-            self.__get_generator_act().send(None)
+            try:
+                self.__get_generator_act().send(self.__message)
+            except StopIteration:
+                self.__handle_stop_iteration()
```
Comment (Adamantios) [inline packages/valory/skills/abstract_round_abci/behaviour_utils.py]: This is more readable:
https://github.com/valory-xyz/open-autonomy/pull/1996#discussion_r1236967696

**Review by Adamantios — APPROVED** [review]

**Pull request merged**